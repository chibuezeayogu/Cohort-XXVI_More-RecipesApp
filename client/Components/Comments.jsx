import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';

const Comments = props => (
  <li className="collection-item avatar z-depth-1">
    <img
      src={props.review.User.imageUrl}
      alt="profile"
      className="circle responsive-img"
    />
    <span className="text-title bold">{props.review.User.firstName}</span>
    <p
      style={{ wordWrap: 'break-word' }}
    >
      {props.review.comment}
    </p>
    <div className="secondary-content">
      { moment(new Date(props.review.createdAt)).fromNow()}
    </div>
  </li>
);

Comments.propTypes = {
  review: PropTypes.shape({
    comment: PropTypes.string,
    createdAt: PropTypes.string,
    User: PropTypes.shape({
      firstName: PropTypes.string,
      imageUrl: PropTypes.string
    })
  }).isRequired
};

export default Comments;
