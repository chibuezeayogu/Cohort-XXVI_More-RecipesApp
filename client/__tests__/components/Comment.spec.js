import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import Comments from '../../Components/Comments';

const props = {
  review: {
    comment: '',
    createdAt: '',
    User: {
      firstName: '',
      imageUrl: ''
    }
  },
};

describe('<Comments />', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Comments {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
