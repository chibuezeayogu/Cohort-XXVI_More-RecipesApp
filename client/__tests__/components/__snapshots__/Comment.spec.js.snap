// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comments /> should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  "node": <li
    className="collection-item avatar z-depth-1"
  >
    <img
      alt="profile"
      className="circle responsive-img"
      src=""
    />
    <span
      className="text-title bold"
    >
      
    </span>
    <p
      style={
        Object {
          "wordWrap": "break-word",
        }
      }
    >
      
    </p>
    <div
      className="secondary-content"
    >
      Invalid date
    </div>
  </li>,
  "nodes": Array [
    <li
      className="collection-item avatar z-depth-1"
    >
      <img
        alt="profile"
        className="circle responsive-img"
        src=""
      />
      <span
        className="text-title bold"
      >
        
      </span>
      <p
        style={
          Object {
            "wordWrap": "break-word",
          }
        }
      >
        
      </p>
      <div
        className="secondary-content"
      >
        Invalid date
      </div>
    </li>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Comments
        review={
          Object {
            "User": Object {
              "firstName": "",
              "imageUrl": "",
            },
            "comment": "",
            "createdAt": "",
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "review": Object {
            "User": Object {
              "firstName": "",
              "imageUrl": "",
            },
            "comment": "",
            "createdAt": "",
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <li
          className="collection-item avatar z-depth-1"
        >
          <img
            alt="profile"
            className="circle responsive-img"
            src=""
          />
          <span
            className="text-title bold"
          >
            
          </span>
          <p
            style={
              Object {
                "wordWrap": "break-word",
              }
            }
          >
            
          </p>
          <div
            className="secondary-content"
          >
            Invalid date
          </div>
        </li>,
        "_debugID": 2,
        "_renderedOutput": <li
          className="collection-item avatar z-depth-1"
        >
          <img
            alt="profile"
            className="circle responsive-img"
            src=""
          />
          <span
            className="text-title bold"
          >
            
          </span>
          <p
            style={
              Object {
                "wordWrap": "break-word",
              }
            }
          >
            
          </p>
          <div
            className="secondary-content"
          >
            Invalid date
          </div>
        </li>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Comments
    review={
      Object {
        "User": Object {
          "firstName": "",
          "imageUrl": "",
        },
        "comment": "",
        "createdAt": "",
      }
    }
  />,
}
`;
